### PR TITLE
More functional test cases

### PR DESCRIPTION
**Pull request by bandinib-amzn**
*Issue #, if available:*

*Description of changes:*

- Adds more functional test cases for role-view
- Adds more functional test cases for role-list
- Adds more functional test cases for user-list
- Adds more functional test cases for tenant-list

Folder level coverage:
File                                                 | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s      
-----------------------------------------------------|---------|----------|---------|---------|--------------------
 public/apps/configuration/panels/role-view          |    92.7 |    76.19 |   86.49 |   93.38 |                        
  cluster-permission-panel.tsx                       |     100 |      100 |     100 |     100 |                        
  index-permission-panel.tsx                         |    96.3 |    83.33 |     100 |    96.3 | 53                     
  role-view.tsx                                      |   86.36 |    58.33 |   58.33 |   87.69 | ...148-149,151-152,371 
  tenants-panel.tsx                                  |     100 |       75 |     100 |     100 | 204-208  
public/apps/configuration/panels/tenant-list        |      88 |    81.25 |   71.88 |   88.78 |                        
  edit-modal.tsx                                     |     100 |      100 |     100 |     100 |                        
  tenant-instruction-view.tsx                        |     100 |      100 |     100 |     100 |                        
  tenant-list.tsx                                    |   86.81 |    78.57 |   67.86 |   87.64 | ...180-223,298,379-380
public/apps/configuration/panels                    |   92.16 |    92.31 |   84.62 |   93.88 |                        
  expression-modal.tsx                               |     100 |      100 |     100 |     100 |                        
  get-started.tsx                                    |   88.89 |      100 |    87.5 |   88.89 | 233                    
  nav-panel.tsx                                      |     100 |      100 |     100 |     100 |                        
  role-list.tsx                                      |   90.24 |       90 |      75 |    92.5 | 189-192 

Coverage before this PR: 67.44%
Coverage after this PR: 72.32%
The change in coverage: +4.88%

Codecov report: https://codecov.io/gh/opendistro-for-elasticsearch/security-kibana-plugin/commits

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by zengyan-amazon — APPROVED** [review]

File: public/apps/configuration/panels/role-view/test/tenants-panel.test.tsx [hunk 1]
```diff
@@ -91,6 +94,139 @@ describe('Role view - tenant panel', () => {
...
+      const columns = component.prop<
+        Array<EuiTableFieldDataColumnType<RoleTenantPermissionDetail>>
+      >('columns');
+      const viewDashboardRenderer = columns[3].render as (tenant: string) => JSX.Element;
+      const Container = (props: { tenant: string }) => viewDashboardRenderer(props.tenant);
+      const result = shallow(<Container tenant={RoleViewTenantInvalidText} />);
+      expect(result).toMatchSnapshot();
+    });
+
+    it('render view visualization column when tenant is invalid', () => {
```
Comment (yuxitang-amzn) [inline public/apps/configuration/panels/role-view/test/tenants-panel.test.tsx:203]: What does it mean by 'tenant is invalid'?

**Review by yuxitang-amzn — APPROVED** [review]
Looks good, huge amount of coverage improvement. Just one small question.

File: public/apps/configuration/panels/role-view/test/tenants-panel.test.tsx (hunk 1 above)
Comment (bandinib-amzn) [reply public/apps/configuration/panels/role-view/test/tenants-panel.test.tsx:203]: It mean that when there are multiple tenants or tenant pattern as tenant permission.

Comment (bandinib-amzn) [reply public/apps/configuration/panels/role-view/test/tenants-panel.test.tsx:203]: I'll change the text. It's confusing.

**Pull request merged**